docs/design/rubygems.md

```diff
@@ -0,0 +1,450 @@
+This option is deprecated in favor of the deployment setting.
+
+There is also `--prefer-local` flag, which will prefer the local cache over the remote source, but it ain't working at all.
+Environment variable `BUNDLE_ALLOW_OFFLINE_INSTALL` is not working either with `bundle install` for some reason,
+which could be probably the most logical solution in this case.
+
+##### Copy
+
+Copy the local configuration file from the user repository to {output_dir} and set BUNDLE_APP_CONFIG to the new location.
+Then, just append all the environment variables needed to the "new" copy of the user configuration file.
+Bundler will rewrite previous values with the new ones when installing gems.
+
+#### Inject
+
+The other solution would be to inject the config file directly and rewrite the values.
+
+### Metadata
+
+### git repository URL
+
+- git repository URL is used in other package managers as well
+- no version information available
+- gems in the repository are path dependencies in the `Gemfile.lock` ?!
+
+### `{gem}.gemspec` file
+
+- the file is optional
+- complete metadata about the gem
+
+Gemfile must contain a _gemspec_ line, + the `{gem}.gemspec` file must be present in the repository.
+Bundle will add the gem as a path dependency to the `Gemfile.lock` file.
+This could be done via gemlock-parser by checking the path.
+
+```ruby
+source "https://rubygems.org"
+
+gemspec
+...
+```
+
+```Gemfile.lock
+...
+PATH
+ remote: .
+ specs:
+ tmp (0.1.2)
+...
+```
+
+### PURL
+
+Examples from [github.com/purl-spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#gem).
+The platform qualifiers key is used to specify an alternative platform, such as java for JRuby.
+
+```txt
+pkg:gem/ruby-advisory-db-check@0.12.4
+pkg:gem/jruby-launcher@1.1.2?platform=java
+```
+
+- **name:** gem name
+- **namespace:** N/A
+- **qualifiers:** vcs_url (GIT dependencies), checksum, platform (ruby) ?
+- **subpath:** subpath from the root (PATH dependencies)
+- **type:** "gem"
+- **version:** gem version
+
+### Offline installation
+
+PoC (Proof of Concept) for offline installation with cachi2 lives in this [branch](https://github.com/slimreaper35/cachi2/tree/poc).
+To achieve offline installation, clone the repository with the `Gemlfile.lock` file and take the following steps:
+
+#### Pre-fetch dependencies
+
+```bash
+cachi2 fetch-deps --source . rubygems
+```
+
+#### Generate environment variables
+
+```bash
+cachi2 generate-env --output cachi2.env --for-output-dir /tmp/cachi2-output cachi2-output
+```
+
+#### Build the image with offline installation
+
+```Dockerfile
+...
+RUN source /tmp/cachi2.env && bundle install
+```
+
+```bash
+podman build . \
+  --volume "$(realpath ./cachi2-output)":/tmp/cachi2-output:Z \
+  --volume "$(realpath ./cachi2.env)":/tmp/cachi2.env:Z \
+  --network none \
+  --tag ruby
+```
+
+### Summary
```

Comment:
We could also add a task to create a custom test repo, [similar](https://github.com/cachito-testing/cachi2-gomod) to what we do with other package managers, trying to explore all the corner cases we can thing of. This should be probably one of the first tasks we tackle.

Reply:
There is 
`add integration and e2e tests` below